Intersect.Client/Entities/Player.cs

```diff
@@ -2406,9 +2409,45 @@ public override void DrawName(Color textColor, Color borderColor, Color backgrou
                 }
             }
 
+            if (Globals.Me.Id != Id)
+            {
+                // Party member names
+                if (Globals.Me.IsInMyParty(this) && CustomColors.Names.Players.TryGetValue("PartyMember", out var partyColors))
+                {
+                    textColor = partyColors.Name;
+                    borderColor = partyColors.Outline;
+                    backgroundColor = partyColors.Background;
+                }
+                // Guildies
+                else if (Globals.Me.IsInGuild && Guild == Globals.Me.Guild && CustomColors.Names.Players.TryGetValue("GuildMember", out var guildColors))
+                {
+                    textColor = guildColors.Name;
+                    borderColor = guildColors.Outline;
+                    backgroundColor = guildColors.Background;
+                }
+
+                // Enemies in PvP
+                if (!Globals.Me.IsAllyOf(this) && Globals.Me.MapInstance.ZoneType != MapZone.Safe && CustomColors.Names.Players.TryGetValue("Hostile", out var hostileColors))
+                {
+                    textColor = hostileColors.Name;
+                    borderColor = hostileColors.Outline;
+                    backgroundColor = hostileColors.Background;
+                }
+            }
+
             DrawNameAndLabels(textColor, borderColor, backgroundColor);
         }
 
+        public override bool IsAllyOf(Player en)
+        {
+            if (base.IsAllyOf(en)) 
+            {
+                return true;
+            }
+
+            return Globals.Me.IsInMyParty(en) || Globals.Me.IsInMyGuild(en) || en.MapInstance.ZoneType == MapZone.Safe;
```

Comment:
Shouldn't the first two conditions be `IsInMyParty()` and `IsInMyGuild()` without the `Globals.Me.` qualification?

The `IsAllyOf()` method is being misused if the `Globals.Me.` qualification.

Reply:
Good catch, fixed